### ISSUE TITLE
Cloud First: Indicate in the Elasticsearch docs which breaking changes affect Cloud

### ISSUE THREAD

**Issue by kilfoyle** [elastic/elasticsearch#66757]
In the Cloud [Upgrade to Elasticsearch X.X](https://www.elastic.co/guide/en/cloud/current/ec-upgrading-v7.html) topics we recommend users to review the [breaking changes in Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html). If possible, it would be ideal if on that breaking changes page we can indicate which breaking changes affect Cloud, similar to how we mark Cloud-affecting user settings with a "Cloud" icon.

**Comment by lockewritesdocs** [comment]
The breaking changes are now listed as part of the upgrading guide for each version, such as the [breaking changes for 7.10](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/migrating-7.10.html#breaking-changes-7.10). However, I think the point about determining which of these changes affect Cloud is still valid. We could do this retroactively for the current docs, but need some process for Developers to validate those changes against Cloud going forward. 

@jasontedor, can we suggest to the ES team that if they're adding a breaking change, to ensure that they're testing on Cloud as well? Or do we assume that if it's a breaking change in ES that it's also a breaking change in Cloud since the latter is built on the former?

**Comment by jasontedor** [comment]
@lockewritesdocs Indeed, it's the latter, I would expect that most breaking changes impact Cloud users in one form or another. There are certainly changes that do not, but I think they're the exception rather than the norm. Given that, I'm not sure if extra notation is needed here?

**Comment by lockewritesdocs** [comment]
@jasontedor, I agree. If it's a breaking change in ES, then it's also in Cloud. Unless @kilfoyle has any major objections, I'll close this issue.

**Comment by kilfoyle** [comment]
Thanks for investigating this! Given that most breaking changes affect Cloud (I didn't know that!), I agree with not implementing this. @nrichers, this came out of a discussion that you and I had, so please let us know if you have thoughts on the matter.

**Comment by nrichers** [comment]
> I would expect that most breaking changes impact Cloud users in one form or another.

@jasontedor I know this issue got closed just now, but I wonder if there's not some nuance to this discussion that is worth exploring more? If so, it might be worth reopening the issue. 

**How do we indicate that there is a breaking ES change that doesn't actually impact Cloud users at all?** 

E.g. the change is technically breaking because it's breaking for Elasticsearch, but ESS deals with it for our users through some additional migration tooling, meaning there is nothing users have to do.

These 'non-breaking breaking' changes are fairly common and the approach by the Stack and Solutions team on Cloud has been that, if a breaking change can be addressed without needing to know about a customer's specific configuration, we do all the legwork required to keep the upgrade a one-click operation. E.g.

* User settings that lead to breaking changes are fairly common, but we handle their configuration as much as possible
* We list https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#tls-v1-removed as a breaking change in our ES docs, but no user on Cloud ever needed to deal with this, as the change was handled by the ESS team. 
 
The same is true for any other breaking changes that a managed service could be expected to handle for its users. 

**How do we indicate breaking changes that affect _only_ Cloud?** 

Rarer, but they do exist. Here is a breaking change we had in our ESS docs that results from an ES security setting that might affect only Cloud users, given that it involves the ESS proxy:

```
* Elasticsearch clusters provisioned prior to January 25th, 2019 have a CORS policy set at the proxy layer. This policy is now deprecated and we will be moving to a model that requires you to configure your own http module CORS policy. The `http.cors.*` settings are <<{p}-add-user-settings,configurable as Elasticsearch user settings>> for all of your deployments.
+
IMPORTANT: If your use case depends on the ability to receive CORS requests and you have an Elasticsearch cluster that was provisioned prior to January 25th 2019, you must manually apply the `http.cors.*` settings before the September 22 deadline, or you will potentially have an operational impact.
```

The bit that would interest me is how and where we indicate these differences in breaking changes, so that users can find them easily. Today, an ES user reviewing breaking changes in the ES docs will have no indication that these differences exist, which goes against the whole cloud-first effort to make the info appear in the right context rather than splitting it across ES and Cloud docs.
